### PR TITLE
[SYCLomatic] Support migration for cub::BlockRadixSort API

### PR DESCRIPTION
**Pull request by abhilash1910**
Add Migration of CUB Block Radix Sort API.
cc @yihanwg

### CODE REVIEW

File: clang/lib/DPCT/ExprAnalysis.cpp [hunk 1]
```diff
@@ -118,8 +118,7 @@ ExprAnalysis::getSpellingOffsetAndLength(SourceLocation Loc) {
     TokenLen = 1;
   }
 
-  return std::pair<SourceLocation, size_t>(
-      Loc, TokenLen);
+  return std::pair<SourceLocation, size_t>(Loc, TokenLen);
```
Comment (yihwang-nv) [inline clang/lib/DPCT/ExprAnalysis.cpp]: Let's revert the unnecessary format change

File: clang/lib/DPCT/CUBAPIMigration.cpp [hunk 2]
```diff
@@ -1143,6 +1146,36 @@ void CubRule::processBlockLevelMemberCall(const CXXMemberCallExpr *BlockMC) {
...
     Repl = NewFuncName + "(" + ParamList + ")";
     emplaceTransformation(new ReplaceStmt(BlockMC, Repl));
+  } else if (FuncName == "Sort" || FuncName == "SortDescending") {
+    if (BlockMC->getMethodDecl()
+            ->getParamDecl(0)
+            ->getType()
+            ->isLValueReferenceType()) {
+      GroupOrWorkitem = DpctGlobalInfo::getItem(BlockMC);
+      if (FuncName == "Sort")
+        {
```
Comment (yihwang-nv) [inline clang/lib/DPCT/CUBAPIMigration.cpp]: Let's format the changed code.

File: clang/test/dpct/cub/blocklevel/blockradixsort.cu [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (yihwang-nv) [inline clang/test/dpct/cub/blocklevel/blockradixsort.cu:5]: Let's revert this change, it's unnecessary.

File: clang/lib/DPCT/CUBAPIMigration.cpp (hunk 2 above)
Comment (abhilash1910) [reply clang/lib/DPCT/CUBAPIMigration.cpp]: Formatted , thanks

File: clang/test/dpct/cub/blocklevel/blockradixsort.cu (hunk 3 above)
Comment (abhilash1910) [reply clang/test/dpct/cub/blocklevel/blockradixsort.cu:5]: Yes I removed this temporarily to see clang format failure in source. Added thanks

**Comment by yihwang-nv** [comment]
@abhilash1910 https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc This experimental feature will make migration easier.

**Comment by abhilash1910** [comment]
Close as implemented.
